app/Http/Controllers/AdminController.php

```diff
@@ -90,4 +91,32 @@ public function question_store(Request $request)
         
         return redirect(route("admin.users"));
     }
+
+    public function update_question(Request $request, $id)
+    {
+        $question = Question::findOrFail($id);
+        $question->update([
+            'question' => $request->question,
+            ]);
+
+        $loop = 0;
+        foreach ($question->choices as $choice) {
+            $loop++;
+            $check = "check" . $loop;
+            if ($request->$check == "on") {
+                $is_correct = true;
+            } else {
+                $is_correct = false;
+            }
+
+            $choice_number = "choice" . $loop;
+            $choice = Choice::findOrFail($choice->id);
+            $choice->update([
+                'choice' => $request->$choice_number,
+                'is_correct' => $is_correct,
+                ]);
+        }
```

Comment:
I think you can do it like this, since foreach can also have the index of the current iteration.

```suggestion
        foreach ($question->choices as $key => $choice) {
            $key++;
            $check = "check" . $key;
            if ($request->$check == "on") {
                $is_correct = true;
            } else {
                $is_correct = false;
            }

            $choice_number = "choice" . $key;
            $choice = Choice::findOrFail($choice->id);
            $choice->update([
                'choice' => $request->$choice_number,
                'is_correct' => $is_correct,
                ]);
        }
```